proof/crefine/ARM/Schedule_C.thy

```diff
@@ -159,6 +171,50 @@ lemma switchToThread_ccorres':
      apply auto
   done
 
+lemma ccorres_pre_getQueue:
+  assumes cc: "\<And>queue. ccorres r xf (P queue) (P' queue) hs (f queue) c"
+  shows   "ccorres r xf (\<lambda>s. P (ksReadyQueues s (d, p)) s \<and> d \<le> maxDomain \<and> p \<le> maxPriority)
+                        {s'. \<forall>queue. (let cqueue = index (ksReadyQueues_' (globals s'))
+                                                         (cready_queues_index_to_C d p) in
+                                      ctcb_queue_relation queue cqueue) \<longrightarrow> s' \<in> P' queue}
+           hs (getQueue d p >>= (\<lambda>queue. f queue)) c"
+  apply (rule ccorres_guard_imp2)
+  apply (rule ccorres_symb_exec_l2)
+     defer
+     defer
+     apply (rule gq_sp)
+    defer
+    apply (rule ccorres_guard_imp)
+    apply (rule cc)
+     apply clarsimp
+     apply assumption
+    apply assumption
+   apply (clarsimp simp: getQueue_def gets_exs_valid)
+  apply clarsimp
+  apply (drule spec, erule mp)
+  apply (erule rf_sr_ctcb_queue_relation)
+   apply (simp add: maxDom_to_H maxPrio_to_H)+
+  done
+
+(* FIXME: replace the version of this rule in Schedule_R with this more general version *)
+lemma lookupBitmapPriority_obj_at':
```

Comment:
Is this an old FIXME that was moved or something new we should resolve now?

Reply:
This is old. I will see how easy it would be do solve this FIXME and will do it if it's not too bad

Reply:
Fixed